apps/admin/src/pages/form/utils/filterFormResponse.ts

```diff
@@ -1,36 +1,76 @@
-const allowLists: Record<string, string[]> = {
-  panel: ['label', 'key', 'type', 'input', 'components'],
-  radio: ['label', 'description', 'key', 'type', 'input', 'values'],
-  select: ['label', 'description', 'key', 'type', 'input', 'widget', 'placeholder', 'data'],
-  selectboxes: ['label', 'description', 'key', 'type', 'input', 'values'],
-  textarea: ['label', 'description', 'key', 'type', 'input', 'autoExpand', 'maxCharCount'],
-  textfield: ['label', 'description', 'key', 'type', 'input'],
+import {
+  $FormCheckboxComponentInput,
+  $FormComponentInputValidate,
+  $FormInput,
+  $FormPanelComponentInput,
+  $FormRadioComponentInput,
+  $FormSelectComponentInput,
+  $FormTextAreaComponentInput,
+  $FormTextFieldComponentInput,
+} from '@meldingen/api-client'
+import filter from 'uber-json-schema-filter'
+
+const filterBySchemaPerType = (obj: any) => {
+  switch (obj.type) {
+    case 'panel':
+      return filter($FormPanelComponentInput, obj)
+    case 'radio':
+      return filter($FormRadioComponentInput, obj)
+    case 'select':
+      return filter($FormSelectComponentInput, obj)
+    case 'selectboxes':
+      return filter($FormCheckboxComponentInput, obj)
+    case 'textarea':
+      // Add autoExpand to object if it doesn't exist, the builder doesn't do that by default
+      if (!obj.hasOwnProperty('autoExpand')) {
+        return filter($FormTextAreaComponentInput, { ...obj, autoExpand: false })
+      }
+
+      return filter($FormTextAreaComponentInput, obj)
+    case 'textfield':
+      return filter($FormTextFieldComponentInput, obj)
+    default:
+      throw Error(`Type ${obj.type} is unknown, please add it to filterFormResponse.`)
+  }
+}
+
+const getFilteredValidateObject = (validateObj: any) => {
+  const validate = filter($FormComponentInputValidate, validateObj)
+
+  // Explicitly remove the 'json' key if its value is an empty string, the API doesn't accept that
+  if (validate?.json === '') {
+    delete validate.json
+  }
+
+  return validate
 }
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export const filterFormResponse = (obj: any): any => {
```

Comment:
Shouldn't it be possible to provide a return type for filterFormResponse?

Reply:
Good catch, this doesn't have to be `any`. I updated it!